assets/dev/js/editor/utils/controls-css-parser.js

```diff
@@ -37,6 +37,34 @@ ControlsCSSParser = elementorModules.ViewModule.extend( {
 		} );
 	},
 
+	updateBackgroundImagesDimension( control, values ) {
+		const backgroundDimensionsPrefix = 'background_dimensions';
+		if ( control.name.includes( 'background_image' ) ) {
+			const imageSize = values[ backgroundDimensionsPrefix ];
+			if ( imageSize ) {
```

Comment:
Same here about the large `if` statement

Reply:
Not relevant anymore, the logic has changed.